gpm_api/tests/io/test_directories.py

```diff
@@ -0,0 +1,152 @@
+                        )
+
+
+def test_get_disk_directory(
+    products: List[str],
+    product_types: List[str],
+    versions: List[int],
+    tmpdir,
+) -> None:
+    """Test that the disk directory is correct."""
+
+    date = datetime.datetime.strptime("2021-01-01", "%Y-%m-%d").date()
+
+    base_dir = os.path.join(tmpdir, "gpm_api_data")
+
+    for product in products:
+        for product_type in product_types:
+            for version in versions:
+                dir_path = dir.get_disk_directory(
+                    base_dir,
+                    product,
+                    product_type,
+                    version,
+                    date,
+                )
+
+                # Work only on product if product_type are compatible
+                if product in available_products(product_type=product_type):
+                    product_category = dir.get_product_category(product)
+                    if product_type == "NRT":
+                        assert "V0" not in dir_path
+                        assert dir_path == os.path.join(
+                            base_dir,
+                            os.path.join("GPM", product_type, product_category, product),
+                            date.strftime("%Y"),
+                            date.strftime("%m"),
+                            date.strftime("%d"),
+                        )
+                    elif product_type == "RS":
+                        assert str(version) in dir_path
+                        # Literal
+                        assert dir_path == os.path.join(
+                            base_dir,
+                            os.path.join(
+                                "GPM", product_type, f"V0{version}", product_category, product
+                            ),
+                            date.strftime("%Y"),
+                            date.strftime("%m"),
+                            date.strftime("%d"),
+                        )
+
+
+def test_get_pps_nrt_product_dir(products: List[str]) -> None:
+    """Test NRT product type folder name
+
+    Depends on gpm_api.io.directories._get_pps_nrt_product_folder_name()
+    """
+
+    date = datetime.datetime(2021, 1, 1).date()
+
+    for product in products:
+        # Only work on NRT products
+        if product in available_products(product_type="NRT"):
+            # Dependent on dir forming private function
+            foldername = dir._get_pps_nrt_product_folder_name(product)
+
+            res = dir._get_pps_nrt_product_dir(product, date)
+            if product in available_products(
+                product_type="NRT",
+                product_category="IMERG",
+            ):
+                assert res == os.path.join(
+                    foldername,
+                    date.strftime("%Y%m"),
+                )
+            else:
+                assert res == foldername
+
+
+# def test_get_pps_directory(
```

Comment:
We could have a YAML file here with the expected path?
What I find nice how having YAML files... is that then we can easily track changes to the file over time ;)